tests/executionspec/spec.rs

```diff
@@ -0,0 +1,400 @@
+                .unwrap())
+            .to_schema(),
+        );
+        let snapshot_name = format!("{}_client", spec.safe_name);
+
+        insta::assert_snapshot!(snapshot_name, client);
+    }
+
+    if let Some(test_spec) = spec.test.as_ref() {
+        let app_ctx = spec
+            .app_context(
+                server.first().unwrap(),
+                spec.env.clone().unwrap_or_default(),
+                mock_http_client.clone(),
+            )
+            .await;
+
+        // test: Run test specs
+        for (i, test) in test_spec.iter().enumerate() {
+            opentelemetry.reset();
+
+            let response = run_test(app_ctx.clone(), test)
+                .await
+                .context(spec.path.to_str().unwrap().to_string())
+                .unwrap();
+
+            let mut headers: BTreeMap<String, String> = BTreeMap::new();
+
+            for (key, value) in response.headers() {
+                headers.insert(key.to_string(), value.to_str().unwrap().to_string());
+            }
+
+            let response: APIResponse = APIResponse {
+                status: response.status().clone().as_u16(),
+                headers,
+                body: serde_json::from_slice(
+                    &hyper::body::to_bytes(response.into_body()).await.unwrap(),
+                )
+                .unwrap_or(serde_json::Value::Null),
+                text_body: None,
+            };
+
+            let snapshot_name = format!("{}_assert_{}", spec.safe_name, i);
+
+            insta::assert_json_snapshot!(snapshot_name, response);
+
+            if test.test_traces {
+                let snapshot_name = format!("{}_assert_traces_{}", spec.safe_name, i);
+                insta::assert_json_snapshot!(snapshot_name, opentelemetry.get_traces().unwrap());
+            }
+
+            if test.test_metrics {
+                let snapshot_name = format!("{}_assert_metrics_{}", spec.safe_name, i);
+                insta::assert_json_snapshot!(
+                    snapshot_name,
+                    opentelemetry.get_metrics().await.unwrap()
+                );
+            }
+        }
+
+        mock_http_client.test_hits(&spec.path);
+    }
+
+    tracing::info!("{} ... ok", spec.path.display());
+}
+
+pub async fn load_and_test_execution_spec(path: &Path) -> anyhow::Result<()> {
+    let opentelemetry = init_opentelemetry();
+
+    // todo
+    // Explicitly only run one test if specified in command line args
+    // This is used by test-convertor to auto-apply the snapshots of incompatible
+    // fail-annotated http specs
+
+    let mut specs = vec![];
+    let contents = fs::read_to_string(path)?;
+    let spec: ExecutionSpec = ExecutionSpec::from_source(path, contents)
+        .await
+        .map_err(|err| err.context(path.to_str().unwrap().to_string()))?;
+    specs.push(spec);
+    specs = ExecutionSpec::filter_specs(specs);
+
+    for spec in specs.into_iter() {
+        test_spec(spec, &opentelemetry).await;
```

Comment:
Why are we using `test_spec` if we are using datatest now?

Reply:
that is the existing assert_spec function.
renamed to test_spec.
using datatest-stable not datatest, both are different.
datatest-stable just runs the  test for each file right.
test_spec is the main thing that runs the test(perform snapshot assertions) for the given spec.

Reply:
specs variable at any time contain only 0 or 1 spec based on skip, only annotations.

Reply:
datatest-stable is doing anything else, just run a test for each file.

Reply:
and seems like we should remove `ExecutionSpec::filter_specs` and storing specs as vec anyway since it's not applicable anymore when using datatest.

I've tested that it works fine with standart cargo options https://doc.rust-lang.org/rustc/tests/index.html#filters